src/main/java/core/basesyntax/SquareSupplier.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class SquareSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomSquare() {
+        return new Square(size.getRandomSize(), color.getRandomColor());
+    }
+}

```

Comment:
Don't use verbs for class/interface names. A class name should be a noun or a noun phrase, not a verb. Consider renaming `SquareSupplier` to something that represents what it is, like `SquareFactory` or `SquareProvider`.